testing/test_terminal.py

```diff
@@ -3067,3 +3067,28 @@ def test_pass():
             "*= 1 xpassed in * =*",
         ]
     )
+
+
+class TestNodeIDHandling:
+    def test_nodeid_handling_windows_paths(self, pytester: Pytester) -> None:
```

Comment:
I just tested this on Windows, on the `main` branch, and it passes... does this fail only on Linux?

Reply:
Hey @nicoddemus ,
The problem specifically occurs under these circumstances as mentioned in the issue:
`The IDs show up correctly in --collect-only, and they show correctly when test_x.py is not in a subfolder, or when pytest is not invoked in the subfolder.`

Reply:
But then the test case should make sure to replicate that scenario, no?

Reply:
Yeah. I'll fix that

Reply:
I believe @dongfangtianyu has already taken care of it: https://github.com/nauman897/pytest/pull/1

Reply:
Thanks @dongfangtianyu for the help with these tests.
I can confirm that:
1. The tests fail with the previous logic in `cwd_relative_nodeid`:
```
=========================== short test summary info ===========================
FAILED test_foo.py::test_x[x\y] - assert False
FAILED test_foo.py::test_x[C:\path] - assert False
FAILED test_foo.py::test_x[\] - assert False
FAILED test_foo.py::test_x[C:\path] - assert False
FAILED test_foo.py::test_x[a::b\] - assert False
============================== 5 failed in 0.11s ==============================
```
2. The tests pass after the patch is applied.

I had to make a minor adjustment to the new tests, as Python 3.11 and later requires passing the `encoding_type` param while calling the `write_text` method.

Reply:
That's cool @nauman897  !


> 2. The tests pass after the patch is applied.
> 
> ```
> test_terminal.py::TestNodeIDHandling::test_nodeid_handling_windows_paths PASSED [100%]
> ```

This is a more intuitive output:
```
test_foo.py::test_x[x/y] FAILED                                          [ 20%]
test_foo.py::test_x[C:/path] FAILED                                      [ 40%]
test_foo.py::test_x[\\] FAILED                                           [ 60%]
test_foo.py::test_x[C:\\path] FAILED                                     [ 80%]
test_foo.py::test_x[a::b/] FAILED                                        [100%]
```